src/app/content/components/Page/searchHighlightManager.ts

```diff
@@ -60,11 +60,10 @@ const selectResult = (services: Services, previous: HighlightProp, current: High
 
   if (firstSelectedHighlight) {
     // If selected result is in the collapsed solution then expand it
-    const solution = (firstSelectedHighlight.elements[0] as HTMLElement)
-      .closest('[data-type="solution"][aria-expanded="false"]');
-    if (solution) {
-      const button = solution.querySelector('button');
-      button?.click();
+    const details = (firstSelectedHighlight.elements[0] as HTMLElement)
+      .closest('[data-type="solution"][aria-expanded="false"]')?.querySelector('details');
+    if (details) {
+      details.open = true;
```

Comment:
You also need to add `.ui-solution-visible` to the solution so the show/hide text is correct.

Reply:
@bethshook 
But `.ui-solution-visible` is getting added.
https://github.com/openstax/rex-web/blob/d0c424b002e226ebc23579b3de5cb08361add1a3/src/app/content/components/Page/contentDOMTransformations.ts#L19-L36
https://github.com/openstax/rex-web/blob/d0c424b002e226ebc23579b3de5cb08361add1a3/src/app/content/components/Page/PageComponent.tsx#L189-L193

I'll change the names from `button` to `summayElement` to make it less misleading.

But maybe i'm gonna refactor the toggling code to make it depended on the state of `open` prop of the `HTMLDetailsElement`.

Reply:
That class is added when details opens for highlighted search results? When I tested on the PR env the show/hide text was wrong.

Reply:
This class was added on click of summary element (button). Not it is added on toggle event. 
`details.open = true;` should trigger that event, so it should work now.